nvc/nvc.go

```diff
@@ -179,3 +175,171 @@ func String2Hash(s string) Hash {
 func (h Hash) String() string {
 	return fmt.Sprintf("%016x", uint64(h))
 }
+
+// Writer is an nvc archive writer.
+type Writer struct {
+	toc []TocEntry
+	w   io.WriteSeeker
+
+	// index keeps track of how many times Create has been called.
+	// Since the table of contents is at the start of the archive,
+	// and all archive member files are after the table of contents, the number of
+	// files must be known ahead of time. Otherwise, writing the table of contents
+	// would result in member file contents being partially overwritten.
+	index int
+}
+
+// NewWriter returns an nvc archive writer that writes to w.
+// length is the number of files that will be placed in the archive.
+// Finalize should be called once all files have been written to the archive (via Create or CreateCompressed).
+func NewWriter(w io.WriteSeeker, length uint32) (Writer, error) {
+	// Start by writing 0s to w until the point at which the first file will start
+	headerLen := 8 + 4 + (24 * length)
```

Comment:
Could we make these local constants or derive them from data structures?

Reply:
The worst part is I actually added those already and just didn't use them

Fixed in b9a3a6ce5191a9f412de3f6b84ea774f9b8fb9e5